components/SearchProvince.js

```diff
@@ -83,6 +90,15 @@ function SearchProvince({ onChooseProvince, onSearchGeo, disabled, value }) {
   return (
     <Box onBlur={() => setTimeout(() => setInputFocus(false), 100)}>
       <HStack spacing="2">
+        <Button
+          colorScheme="gray"
+          color="gray.600"
+          style={{ display: backButton === true ? 'block' : 'none' }}
+          aria-label="Back to homepage"
+          onClick={() => (window.location.href = '/')}
+        >
+          {<ArrowBackIcon w={5} h={5} m={-1} />}
+        </Button>
```

Comment:
Hmmm. It's better to go with conditional component rendering like this:
```js
{backButton && <Button {...} />}
```
Don't forget to remove the conditional styling.

And don't use `window.location.href` for redirection. Please use Next.js router instead. See [here](https://nextjs.org/docs/api-reference/next/router).

Reply:
ok, understood.
quick question, do you want to put `router.push()` inside the onClick or create a new function to handle the onClick?

Reply:
I prefer to make a new function to handle the on click.

Reply:
sorry, dumb question. I accidentally push my changes on `package.json` and `.gitigonre` on my commit (I made changes because my node is in 15 and pnpm didn't want to install at the beginning), how to exclude that from the pushed commit?

Reply:
should I create a new PR?